ocp_resources/multi_network_policy.py

```diff
@@ -0,0 +1,56 @@
+from ocp_resources.resource import NamespacedResource
+
+
+class MultiNetworkPolicy(NamespacedResource):
+    """
+    MultiNetworkPolicy object.
+
+    API reference:
+    https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-multi-network-policy.html
+    """
+
+    api_group = NamespacedResource.ApiGroup.K8S_CNI_CNCF_IO
+
+    def __init__(
+        self,
+        network_name=None,
+        policy_types=None,
+        ingress=None,
+        egress=None,
+        pod_selector=None,
+        **kwargs,
+    ):
+        """
+        Create and manage MultiNetworkPolicy
+
+        Args:
+            network_name (str): The name of the NetworkAttachmentDefinition that the policy will impact.
+            policy_types (list, optional): One or both of the valid ip policies - "Ingress" and "Egress".
+            ingress (list, optional): list containing a dictionary specifying the allowed "from" parameters.
+            egress (list, optional): list containing a dictionary specifying the allowed "to" parameters.
+            pod_selector (dict, optional): Map a label to match.
+        """
+        super().__init__(**kwargs)
+        self.network_name = network_name
+        self.policy_types = policy_types
+        self.pod_selector = pod_selector
+        self.ingress = ingress
+        self.egress = egress
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            if not self.network_name:
+                raise ValueError("network_name is required")
+
+            self.res["metadata"]["annotations"] = {
+                f"{self.api_group}/policy-for": f"{self.namespace}/{self.network_name}"
+            }
+            self.res["spec"] = {
+                "podSelector": self.pod_selector or {},
+                "policyTypes": self.policy_types,
```

Comment:
is `self.policy_types` mandatory?

Reply:
yes, you must specify the network the policy will be enforced on.

Reply:
no

Reply:
So why assign it without check?

Reply:
Not sure I understand what you mean. This is a valid policy:
apiVersion: k8s.cni.cncf.io/v1beta1
kind: MultiNetworkPolicy
metadata:
  annotations:
    k8s.v1.cni.cncf.io/policy-for: flat-overlay/flat-l2-nad-vma-vmb 
  name: ingress-mnp
  namespace: flat-overlay                         
spec:
  podSelector: {}
  policyTypes: null

So this field doesn't require checking before assigning.

Reply:
yes it is, if the user do not send it do no assign it